src/leiningen/ring/uberwar.clj

```diff
@@ -11,27 +14,45 @@
 (defn get-classpath [project]
   (if-let [get-cp (resolve 'leiningen.core.classpath/get-classpath)]
     (get-cp project)
-    (->> (:library-path project) io/file .listFiles (map str))))
-
-(defn contains-entry? [^java.io.File file ^String entry]
-  (with-open [jar-file (JarFile. file)]
-    (some (partial = entry)
-          (map #(.getName ^JarEntry %)
-               (enumeration-seq (.entries jar-file))))))
+    (->> (:library-path project) (io/file) .listFiles (map str))))
 
 (defn jar-dependencies [project]
   (for [pathname (get-classpath project)
-        :let [file (io/file pathname)
-              fname (.getName file)]
-        :when (and (.endsWith fname ".jar")
-                   ;; Servlet container will have it's own servlet-api impl
-                   (not (contains-entry? file "javax/servlet/Servlet.class")))]
-    file))
-
-(defn jar-entries [war project]
+        :when (str/ends-with? pathname ".jar")]
+    (io/file pathname)))
+
+(def ^:private javax-servlet?  #{"javax/servlet/Servlet.class"})
+(def ^:private pom-properties? #(str/ends-with? % "pom.properties"))
+
+(defn- war-path-for-jar
+  [^java.io.File jar]
```

Comment:
Arguments should fit on the same line as the function name where there's room to do so.